src/snowcli/cli/project/definition_manager.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import os
+from pathlib import Path
+from typing import List
+import functools
+from typing import Optional
+from snowcli.exception import MissingConfiguration
+from snowcli.cli.project.definition import load_project_definition
+
+
+class DefinitionManager:
+    BASE_CONFIG_FILENAME = "snowflake.yml"
+    USER_CONFIG_FILENAME = "snowflake.local.yml"
+
+    project_root: Path
+    _project_config_paths: List[Path]
+
+    def __init__(self, project: str | None = None) -> None:
+        search_path = os.getcwd()
+        if project and len(project) > 0:
+            search_path = os.path.abspath(project)
+        config_files = self._find_config_files(Path(search_path))
+        if not config_files:
+            raise MissingConfiguration(
+                f"Cannot find native app project configuration. Please provide a path to the project or run this command in a valid native app project directory."
```

Comment:
If I understand correctly, then the `project` package is designed to be used not only by native apps, so exception messages shouldn't have native apps context (at least not hardcoded).